client-spark/spark-2/src/main/java/org/apache/spark/shuffle/celeborn/SparkShuffleManager.java

```diff
@@ -66,6 +67,26 @@ public class SparkShuffleManager implements ShuffleManager {
   private ExecutorShuffleIdTracker shuffleIdTracker = new ExecutorShuffleIdTracker();
 
   public SparkShuffleManager(SparkConf conf, boolean isDriver) {
+    int maxStageAttempts =
+        conf.getInt(
+            "spark.stage.maxConsecutiveAttempts",
+            DAGScheduler.DEFAULT_MAX_CONSECUTIVE_STAGE_ATTEMPTS());
+    int maxTaskAttempts = (Integer) conf.get(package$.MODULE$.MAX_TASK_FAILURES());
```

Comment:
nit: Move this into the common utils, so that both `getMapAttemptNumber` and `validateAttemptConfig` (or some such) are together. Also, we can use the same for both spark 2 and 3.

Reply:
In Spark 2, the configuration parameter `spark.task.maxFailures` is referred to as `MAX_TASK_FAILURES`, while in Spark 3, it is renamed to `TASK_MAX_FAILURES`. As a result, we cannot utilize the same `validateAttemptConfig` for both Spark 2 and Spark 3.  I have added a SparkCommonUtils class that includes the methods `validateMaxAttempts` and `getEncodedAttemptNumber`

Reply:
Sounds good to me.
Note that we can directly use `spark.task.maxFailures` instead (the default of `4`, if key is absent, is not going to change).